### DISCUSSION TITLE
'S3FullAccessPolicy' added to lambda function is not allowing to put object to s3 bucket.

### DISCUSSION THREAD

**Discussion by RZ-Jyothsna** [aws/serverless-application-model#3299] [Questions] [answered]

**Comment by RZ-Jyothsna** [comment]
In my SAM application, I need to add file to s3 bucket . I am doing it with "upload_fileobj" function. I have addded "S3FullAccessPolicy" to this lambda to add permissions. But I see this error

[ERROR] ClientError: An error occurred (AccessControlListNotSupported) when calling the PutObject operation: The bucket does not allow ACLs


My template file:
![Screenshot from 2023-08-11 12-23-00](https://github.com/aws/serverless-application-model/assets/117725492/2465a58a-290b-4248-b6b4-33fb6d400003)

I also read that adding acl is not recommended in this documentation -'https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket.html#cfn-s3-bucket-accesscontrol'

Is it possible to solve this without adding ACL for s3.

**Comment by ssenchenko** [comment, marked as answer]
As you can see in the [S3FullAccess policy](https://github.com/aws/serverless-application-model/blob/843a6fc1388ae2e39bdbcfbdf2fbc32f9ce1e759/samtranslator/policy_templates_data/policy_templates.json#L1742C7-L1742C7) it does have `Get/PutObjectAcl` action items and we keep it there mainly because of the backwards compatibility cases. Using `*ObjectAcl` permissions is indeed not recommended because of the security concerns.

Though you can browse the aforementioned file and find the policy which suits your needs better, I encourage you to use a **Connector** instead if you're writing to S3 from Lambda function. If it's the case, you'll need a  Lambda to S3 connector. You can read more about connectors [here](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-resource-connector.html). And the list of already existing profiles is [here](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/reference-sam-connector.html).
